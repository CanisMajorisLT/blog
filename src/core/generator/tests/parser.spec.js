const blogPostExample = require('../../../../tests/fixtures/blogPostExample.js');
const { parseArticle } = require('../parser.ts');
const { getArticles } = require('../fileFinder.ts');

describe('parser#parseArticle', () => {
    it('parses correct document', () => {
        expect(parseArticle(blogPostExample)).toMatchSnapshot();
    });
});

describe('fileFinder#getArticles', () => {
    it('returns an array of matched files at given glob pattern', async() => {
        const articles = await getArticles('tests/**/*.md');
        expect(articles).toMatchSnapshot();
    });
});
