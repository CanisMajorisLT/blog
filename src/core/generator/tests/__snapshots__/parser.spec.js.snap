// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fileFinder#getArticles returns an array of matched files at given glob pattern 1`] = `
Array [
  "---
Title:  Test Article
Author: Robot Mr. Bot
Date:   2017-01-01
tags:
- interesting text
- very good content
- smart
- navigation
---

# Hello

One

## World

Two
",
]
`;

exports[`parser#parseArticle parses correct document 1`] = `
Object {
  "articleHtml": "<hr />
<p>Title:  Test Article<br />
Author: Robot Mr. Bot<br />
Date:   2017-01-01<br />
tags:</p>
<ul>
<li>interesting text</li>
<li>very good content</li>
<li>smart</li>
<li>navigation</li>
</ul>
<hr />
<h1>Hello</h1>
<p>One</p>
<h2>World</h2>
<p>Two</p>
",
  "articleJson": Object {},
  "metaData": Object {
    "Author": "Robot Mr. Bot",
    "Date": 2017-01-01T00:00:00.000Z,
    "Title": "Test Article",
    "tags": Array [
      "interesting text",
      "very good content",
      "smart",
      "navigation",
    ],
  },
}
`;
